### PR TITLE
[react-markings] Add types

### PR DESCRIPTION
**Pull request by Methuselah96**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/react-markings/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    type ImageProps = CommonmarkReactRenderer.ImageProps;
+    type ListProps = CommonmarkReactRenderer.ListProps;
+    type Renderers = CommonmarkReactRenderer.Renderers;
+
+    interface Options {
+        renderers?: Partial<Renderers>;
+    }
+
+    type Markings = (...strings: Array<TemplateStringsArray | React.ReactElement>) => React.ReactElement;
+    type Customize = (opts: Options) => Markings;
```
Comment (sheetalkamat) [inline types/react-markings/index.d.ts:26]:
```suggestion
    export function customize(opts: Options): Markings;
```

File: types/react-markings/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+interface md {
+    (...strings: Array<TemplateStringsArray | React.ReactElement>): React.ReactElement;
+    customize: md.Customize;
+}
+
+declare const md: md;
```
Comment (sheetalkamat) [inline types/react-markings/index.d.ts:34]:
```suggestion
function md(...strings: Array<TemplateStringsArray | React.ReactElement>): React.ReactElement;
```

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/react-markings/index.d.ts (hunk 2 above)
Comment (Methuselah96) [reply types/react-markings/index.d.ts:34]: I'm not sure I understand this change. Would you still be able to call both `` md`Stuff here` `` and `` md.customize(options)`Stuff here` ``?

Comment (Methuselah96) [reply types/react-markings/index.d.ts]: @sheetalkamat Thanks! Addressed.

**Pull request merged**